Export structurizr

mh182 opened the pull request:
This is a draft PR to discuss the implementation for #1016.

mh182 committed:
Add boilerplate code for exportStructurizr

This is by no means ready to integration. It is a hack which shows the
prototype how do extract the diagrams with `structurizr-cli`.

mh182 committed:
refactor (exportStructurizr): use structurizr libraries instead of gradle plugin

Instead of using the Structurizr CLI gradle plugin we directly use the
Structurizr libraries (see https://github.com/structurizr/cli/discussions/106#discussioncomment-5026045).

Integrates the code provided with https://github.com/docToolchain/arc42-c4-software-architecture-documentation-example/commit/647b407a2a31fb9da627e13dd025f6d647d7b2ba

mh182 committed:
refactor (exportStructurizr): provide configuration parameters

mh182 committed:
Add test case ExportStructurizrSpec (failed)

Note: the test case fails which has to be resolved before we integrate
it upstream.

rdmueller committed:
added (ADR): ADR01-Structurizr

rdmueller committed:
add (generateSite): FENCED_CODE_BLOCKS for markdown (#1039)

rdmueller committed:
release (dtcw): updated version in wrapper (#1041)

rdmueller commented on scripts/exportStructurizr.gradle:
```diff
@@ -0,0 +1,70 @@
+        description: 'exports the views of a Structurizr DSL file to diagramms'
+) {
+
+    def workspaceFile = findProperty("structurizr.workspace")?:config.structurizr.workspace
```
that's great that it will also work with config properties!

rdmueller reviewed (approved):
looks good!

mh182 committed:
fix (dtcw): JAVA_HOME unbound variable (closed #1043) (#1044)

rdmueller committed:
put 0-params check in front of all other checks

rdmueller committed:
increased version of dtcw

rdmueller committed:
Update 020_run_tests.adoc

Commit:
Use the configured proxy for the label API calls

If a proxy is necessary for the current environment it needs to be used
here as well or the publish job will fail.

rdmueller committed:
change (publishToConfluence): Use the configured proxy for the label API calls (#1052)

If a proxy is necessary for the current environment it needs to be used
here as well or the publish job will fail.

Co-authored-by: Erlwein, Björn <bjoern.erlwein@huk-coburg.de>

mh182 committed:
Add configuration parameter for output format

rdmueller committed:
fix (publishToConfluence): broken (#1055)

mh182 committed:
Polish implementation so it is ready for release

mh182 commented:
I think I have finished the implementation part of this ticket. 
Now I have the following questions:
* Cooperation workflow: I rebased my local git branch (not yet pushed) to the `ng` branch. This means I would force push the changes to my fork [`mh182:exportStructurizr`](https://github.com/mh182/docToolchain/tree/exportStructurizr), which this review is based on.
    * How do we make a new review-cycle?
    * Is this the right way how to do review cycles on the same PR?
* I started writing the task documentation. How can I generate the documentation shown on http://doctoolchain.org locally on my machine? The developer documentation provides how it is done via the GitHub interface, which is not the right way to do it when I develop.

mh182 commented:
I screwed up this PR. My branch is rebased on ng and includes also the ADR.
I think I will just close this PR and create a new one agains the ng branch. 
My question is how do we handle the ADR (which IMHO is not ready yet)?

rdmueller commented:
* rebase can also be done through the github ui
* if the ADR is part of the PR, not problem. It is marked as draft
* generation of docs should be just `./dtcw generateSite`

rdmueller commented:
and regarding the docs - the last netlify task in a PR is a link to preview-docs
